README.md

```diff
@@ -0,0 +1,130 @@
+
+This document intends to give a brief introduction to the Highlight API and how to use it, as well as compare performance and ease-of-use with a traditional way of highlighting text.
+
+## How to use it
+
+Custom highlights are represented by [Highlight](https://drafts.csswg.org/css-highlight-api-1/#creation) objects. A Highlight contains ranges that can be added either by passing them to its constructor or by setting them to the object after creation. For example:
+
+```Javascript
+let range1 = new Range();
+let range2 = new StaticRange({startContainer: document.body, startOffset: 0, endContainer: document.body, endOffset: 1});
+let highlight = new Highlight(range1);
+highlight.add(range2);
+```
+
+In order to have any effect, custom highlights need to be registered into the highlight registry, represented by a [HighlightRegistry](https://drafts.csswg.org/css-highlight-api-1/#registration) object. The highlight registry is accessed via the `highlights` attribute of the CSS namespace, and represents all the custom highlights registered for the current global object’s associated Document. It is a map that relates custom highlight names to Highlight objects. The names are represented by Strings, and they are used to link a custom highlight to its [corresponding style](https://drafts.csswg.org/css-highlight-api-1/#styling-highlights) defined in the style section. For example:
+
+```Javascript
+<style>
+  ::highlight(foo){
+      color: blue;
+  }
+</style>
+<script>
+  let customHighlight = new Highlight();
+  CSS.highlights.set('foo', customHighlight);
+</script>
+```
+
+In the case above, a custom highlight is registered under the name 'foo', so all ranges contained in it will be styled with text color blue, as specified in the style section.
+
+Finally, the next example shows a simple way of highlighting selected text with the Highlight API:
+
+```Javascript
+<style>
+  ::highlight(example-highlight) {
+    color: blue;
+  }
+</style>
+
+<script>
+  const highlightName = "example-highlight";
+
+  function highlightSelectionWithAPI() {
+    let range = window.getSelection().getRangeAt(0);

```

Comment:
```suggestion
    let range = window.getSelection().getRangeAt(0).cloneRange();
```

We probably want to clone the range because per https://w3c.github.io/selection-api/#dom-selection-getrangeat the range returned by `getRangeAt()` is a reference, not a copy, of the one held inside the Selection implementation. It is a little suspicious to use that same range inside the Highlight, since a change that range would update both the user selection and the application of the ::highlight style.

In practice it basically works as-is because Selection seems to throw away its internal range whenever the user moves the selection. But I'd like to remove the ambiguity from the demo.

Reply:
Oh yes, I agree, good catch there, thanks!